openmls/src/tree/hashes.rs

```diff
@@ -205,14 +210,16 @@ impl RatchetTree {
                 tree.nodes[index].node = Some(parent_node);
                 // Calculate the parent hash of the current node and return it
                 let parent_hash = &tree.nodes[index].node.as_ref().unwrap().parent_hash;
-                ParentHashInput::new(tree, index, former_index_sibling, parent_hash.as_slice())
-                    // FIXME: remove unwrap.
-                    // It is ok to use `unwrap()` here, since we can be sure the node is not blank
-                    .unwrap()
-                    .hash(tree.ciphersuite, backend)
+                Ok(
+                    ParentHashInput::new(tree, index, former_index_sibling, parent_hash.as_slice())
+                        // FIXME: remove unwrap.
+                        // It is ok to use `unwrap()` here, since we can be sure the node is not blank
+                        .unwrap()
```

Comment:
Is there a particular reason we're not propagating the error here?

Reply:
I really only touched the code in `/tree` where necessary. If you think this won't get overwritten by the tree rewrite I can always take this further.